extensions/gitpod/src/remoteConnector.ts

```diff
@@ -379,22 +413,96 @@ export default class LocalApp extends Disposable {
 		}
 	}
 
-	public async handleUri(uri: vscode.Uri) {
-		if (uri.path === LocalApp.AUTH_COMPLETE_PATH) {
-			this.logger.info('auth completed');
-			return;
+	private async getWorkspaceSSHDestination(workspaceId: string, gitpodHost: string): Promise<string> {
+		const session = await vscode.authentication.getSession(
+			'gitpod',
+			['function:getWorkspace', 'function:getOwnerToken', 'function:getLoggedInUser', 'resource:default'],
+			{ createIfNone: true }
+		);
+
+		const serviceUrl = new URL(gitpodHost);
+
+		const workspaceInfo = await withServerApi(session.accessToken, serviceUrl.toString(), service => service.server.getWorkspace(workspaceId), this.logger);
+		if (workspaceInfo.latestInstance?.status?.phase !== 'running') {
+			throw new NoRunningInstanceError(workspaceId);
 		}
-		this.logger.info('open workspace window: ' + uri.toString());
-		const params: SSHConnectionParams = JSON.parse(uri.query);
-		let resolvedConfig: LocalAppConfig | undefined;
+
+		const workspaceUrl = new URL(workspaceInfo.latestInstance.ideUrl);
+
+		const sshHostKeyEndPoint = `https://${workspaceUrl.host}/_ssh/host_keys`;
+		const sshHostKeyResponse = await fetch(sshHostKeyEndPoint);
+		if (!sshHostKeyResponse.ok) {
+			// Gitpod SSH gateway not configured
+			throw new NoSSHGatewayError(gitpodHost);
+		}
+
+		const sshHostKeys: { type: string, host_key: string }[] = await sshHostKeyResponse.json();
+
+		const ownerToken = await withServerApi(session.accessToken, serviceUrl.toString(), service => service.server.getOwnerToken(workspaceId), this.logger);
+
+		const sshDestInfo = {
+			user: `${workspaceId}#${ownerToken}`,
```

Comment:
This is send to the remote-ssh extension so it has to be this way if not it will ask you to input the password manually, it's the same as pasting the use@host in the input box  when invoking `Connect to Host command` in the remote ssh-extension